### PR TITLE
feature/metrics-log-annotation

### PR DESCRIPTION
**Pull request by gratcliff**
Log decoration - preliminary work:

- Building out 'findOperation' class method on OAS. Enables ingestion of an log, and combines it with the existing OAS spec to determine the target path

### CODE REVIEW

File: src/oas.js [hunk 1]
```diff
@@ -101,6 +102,80 @@ function normalizedUrl(oas) {
   return ensureProtocol(url);
 }
 
+function normalizePath(path) {
+  const curlBacketMatch = /{(.*?)}/;
+
+  return path
```
Comment (dok) [inline src/oas.js]: How about something like this?

```js
'/users/{user_id}/filter_sets/{filter_set_id}'.replace(/{(.*?)}/g, ':$1')
```

Then we can same a little bit of time on iteration

File: src/oas.js [hunk 2]
```diff
@@ -128,6 +203,23 @@ class Oas {
     const operation = getPathOperation(this, { swagger: { path }, api: { method } });
     return new Operation(this, path, method, operation);
   }
+
+  findOperation(logInput) {
+    const { url, method } = logInput.request.log.entries[0].request;
```
Comment (dok) [inline src/oas.js]: We're gonna move away from referencing `.request.log.entries` (we've "exploded" the request object in the HAR object to the root level of the mongo document)!  Let's use `logInput.url` & `logInput.method`

File: src/oas.js [hunk 3]
```diff
@@ -101,6 +102,80 @@ function normalizedUrl(oas) {
...
+    .map(p => {
+      const pathMatch = curlBacketMatch.exec(p);
+      if (pathMatch) return `:${pathMatch[1]}`;
+      return p;
+    })
+    .join('/');
+}
+
+function generatePathMatches(paths, pathName, origin) {
+  return Object.keys(paths)
```
Comment (dok) [inline src/oas.js]: What happens when there are multiple matches here? Can there be multiple matches in this step?

File: src/oas.js (hunk 1 above)
Comment (gratcliff) [reply src/oas.js]: Nice, that's a good improvement!

File: src/oas.js (hunk 2 above)
Comment (gratcliff) [reply src/oas.js]: Has that been updated in that repo yet?

File: src/oas.js (hunk 3 above)
Comment (gratcliff) [reply src/oas.js]: " it deals with multiple matches by prioritizing the match with the least path Parmas. Those wildcards really are the ones that would contribute to finding dups 
If there are other dups (like repeat paths) that would just be an incorrect oas file to begin with" via our slack

File: src/oas.js (hunk 2 above)
Comment (gratcliff) [reply src/oas.js]: Good catch, I'll swap that out! Thanks @dok

Comment (erunion) [reply src/oas.js]: I'd also suggest supplying `findOperation` with the url and method upfront. This package shouldn't need to know about how we store logs in the DB, just that it needs to look for an operation that match a specific pattern.

Comment (gratcliff) [reply src/oas.js]: Good call @erunion, i'll move it to the FE.

File: __tests__/fixtures/local-link.json [hunk 4]
```diff
@@ -0,0 +1,369 @@
...
+    {
+      "url": "http://local-link.com"
+    }
+  ],
+  "paths": {
+    "/2.0/users/{username}": {
+      "get": {
+        "operationId": "getUserByName",
+        "security": [
+					{
```
Comment (erunion) [inline __tests__/fixtures/local-link.json]: Can you fix the weird indents going on here?

**Pull request merged**